crl/updater/updater.go

```diff
@@ -279,6 +287,30 @@ func (cu *crlUpdater) tickIssuer(ctx context.Context, atTime time.Time, issuerNa
 	return nil
 }
 
+// tickShardWithRetry calls tickShard repeatedly (with exponential backoff
+// between attempts) until it succeeds or the max number of attempts is reached.
+func (cu *crlUpdater) tickShardWithRetry(ctx context.Context, atTime time.Time, issuerNameID issuance.IssuerNameID, shardIdx int, chunks []chunk) error {
+	crlID := crl.Id(issuerNameID, crl.Number(atTime), shardIdx)
+
+	var err error
+	for i := 0; i < cu.maxAttempts; i++ {
+		// core.RetryBackoff always returns 0 when its first argument is zero.
+		sleepTime := core.RetryBackoff(i, time.Second, time.Minute, 2)
+		if i != 0 {
+			cu.log.Errf(
+				"Generating CRL failed, will retry in %vs: id=[%s] err=[%s]",
+				sleepTime.Seconds(), crlID, err)
+		}
+		cu.clk.Sleep(sleepTime)
+
+		err = cu.tickShard(ctx, atTime, issuerNameID, shardIdx, chunks)
+		if err == nil {
+			break
```

Comment:
The number of attempts itself is only an issue inasmuch as it a) may make the process take longer (which we measure), or b) may cause extra load on the SA and database (which we measure). Since I don't think we'd ever alert on this metric, and we can get the more directly-relevant metrics already, I think we're good here.